app/code/Magento/Inventory/Model/Indexer/StockItem.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\Indexer;
+
+use Magento\Inventory\Model\Indexer\StockItem\DataProvider;
+use Magento\Inventory\Model\Indexer\StockItem\DimensionFactory;
+use Magento\Inventory\Model\Indexer\StockItem\IndexHandler;
+use Magento\Inventory\Model\Indexer\StockItem\TemporaryIndexHandler;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * @inheritdoc
+ */
+class StockItem implements StockItemIndexerInterface
+{
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var DimensionFactory
+     */
+    private $dimensionFactory;
+
+    /**
+     * @var IndexHandler
+     */
+    private $handler;
+
+    /**
+     * @var TemporaryIndexHandler
+     */
+    private $temporaryHandler;
+
+    /**
+     * @var DataProvider
+     */
+    private $dataProvider;
+
+    /**
+     * StockItem constructor.
+     * @param DimensionFactory $dimensionFactory
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        DimensionFactory $dimensionFactory,
+        StoreManagerInterface $storeManager,
+        IndexHandler $handler,
+        TemporaryIndexHandler $temporaryHandler,
+        DataProvider $dataProvider
+    ) {
+        $this->storeManager = $storeManager;
+        $this->dimensionFactory = $dimensionFactory;
+        $this->handler = $handler;
+        $this->temporaryHandler = $temporaryHandler;
+        $this->dataProvider = $dataProvider;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function executeFull()
+    {
+        $storeIds = array_keys($this->storeManager->getStores());
+
+        foreach ($storeIds as $storeId) {
+            $dimension = [$this->dimensionFactory->create(['name' => 'scope', 'value' => $storeId])];
+            $this->temporaryHandler->cleanIndex($dimension);
+            $this->temporaryHandler->saveIndex($dimension, $this->dataProvider->fetchDocuments([]));
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function executeRow($id)
+    {
+        $this->executeList([$id]);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function executeList(array $ids)
+    {
+        $storeIds = array_keys($this->storeManager->getStores());
+
+        foreach ($storeIds as $storeId) {
+            $dimension = [$this->dimensionFactory->create(['name' => 'scope', 'value' => $storeId])];
```

Comment:
In the documentation  https://github.com/magento-engcom/magento2/wiki/New-Indexer-for-StockItems#index-dimensions stated 

 
>That's why it's proposed to split indexes by the Scope. Taking into account that we have 1 - 1 relation between Scope (Sales Channel) and Stock 
For example, having 5 websites with ids (1 .. 5) we will end up creating 5 different index tables (with the same structure):
```
inventory_stock_item_stock_1
inventory_stock_item_stock_2
inventory_stock_item_stock_3
inventory_stock_item_stock_4
inventory_stock_item_stock_5
```

Current code where dimension is StoreId is copy-pasted from Search Indexer